adafruit_mcp3xxx/adafruit_mcp3xxx.py

```diff
@@ -0,0 +1,248 @@
+
+    def _read_pin_volts_differential(self, diff_pin, voltage):
+        """Reads a MCP3008 differential pin value, returns the voltage as a floating point value."""
+        v_in = self._read(diff_pin, self.pin_count, is_differential=True)
+        return (v_in * voltage) / 1023
+
+class MCP3004(MCP3xxx):
+    """MCP3004 10-bit analog to digital converter instance.
+
+    mcp = adafruit_mcp3xxx.MCP3004(spi,cs)
+    """
+    def __init__(self, spi_bus, cs):
+        super(MCP3004, self).__init__(spi_bus, cs)
+        self.pin_count = 3 #MCP3004 has 4channels.
+        self.type = MCP_3004
+
+    def _read_pin(self, pin):
+        """Reads a MCP3004 pin, returns the value as an integer."""
+        return self._read(pin, self.pin_count)
+
+    def _read_pin_volts(self, pin, voltage):
+        """Reads a MCP3004 pin, returns the voltage as a floating point value."""
+        v_in = self._read(pin, self.pin_count)
+        return (v_in * voltage) / 1023
+
+    def _read_pin_differential(self, diff_pin):
+        """Reads a MCP3004 differential pin value, returns the value as an integer."""
+        return self._read(diff_pin, self.pin_count, is_differential=True)
+
+    def _read_pin_volts_differential(self, diff_pin, voltage):
+        """Reads a MCP3004 differential pin value, returns the voltage as a floating point value."""
+        v_in = self._read(diff_pin, self.pin_count, is_differential=True)
+        return (v_in * voltage) / 1023
+
+class AnalogIn():
+    """AnalogIn for single-ended ADC readings.
+
+    adc = adafruit_mcp3xxx.AnalogIn(mcp, pin)
+
+    :param adc: mcp3xxx object.
+    :param pin: mcp3xxx analog pin.
+    """
+    def __init__(self, adc, pin):
+        self._adc = adc
+        self._pin = pin
+
+    @property
+    def value(self):
+        """Returns the value of an ADC pin as an integer.
+        """
+        return self._adc._read_pin(self._pin)
+
+    @property
+    def volts(self):
+        """Returns the voltage from an ADC pin as a floating point value.
+        """
+        return self._adc._read_pin_volts(self._pin, self._adc.max_voltage)
+
+
+class AnalogIn_Differential():
+    """Reads the difference between two signals
+
+    adc = adafruit_mcp3xxx.AnalogIn_Differential(mcp, pin_1, pin_2)
+
+    :param adc: mcp3xxx object.
+    :param pin_1: mcp3xxx analog pin 1.
+    :param pin_2: mcp3xxx analog pin 2.
+    """
+
+    def __getitem__(self, key):
+        return self._channels[MCP3008_DIFF_PINS[key]]
+
+    def __init__(self, adc, pin_1, pin_2):
+        self._adc = adc
+        self._channels = []
+        self._pin_1 = pin_1
+        self._pin_2 = pin_2
+
+    @property
+    def value(self):
+        """Returns the value from a differential read across two pins as an integer.
+        """
+        diff_pin = MCP3008_DIFF_PINS.get((self._pin_1, self._pin_2), "Difference pin not found.")
+        return self._adc._read_pin_differential(diff_pin)
+
+    @property
+    def volts(self):
```

Comment:
`voltage` not `volts` please. https://circuitpython.readthedocs.io/en/3.x/docs/design_guide.html#sensor-properties-and-units